spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/VisionPredictor.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.optim
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.dataset.{Sample, SampleToMiniBatch}
+import com.intel.analytics.bigdl.models.utils.ModelBroadcast
+import com.intel.analytics.bigdl.nn.SpatialShareConvolution
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.transform.vision.image.{DistributedImageFrame, ImageFeature, ImageFrame}
+
+import scala.reflect.ClassTag
+
+/**
+ * VisionPredictor is a predictor for images
+ *
+ * @param model prediction model
+ * @param postProcessor post process after model forward
+ */
+class VisionPredictor[T: ClassTag](
```

Comment:
Why don't you add a predict(ImageFrame) method to Predictor?

And we definitely need a Module.predict(ImageFrame) method

Reply:
@jason-dai   is this  good to add the `predict` method to `AbstractModule` as this is specific to Image ?

Reply:
Image is the major use case for deep learning today